pkg/virt-launcher/virtwrap/postcopy.go

```diff
@@ -0,0 +1,66 @@
+package virtwrap
+
+import (
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/log"
+	cmdclient "kubevirt.io/kubevirt/pkg/virt-handler/cmd-client"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/cli"
+
+	libvirt "libvirt.org/libvirt-go"
+)
+
+func (l *LibvirtDomainManager) registerIterationEventForPostCopy(vmi *v1.VirtualMachineInstance, dom cli.VirDomain, options *cmdclient.MigrationOptions, migrationError chan error) {
+	l.virConn.DomainEventMigrationIterationRegister(func(c *libvirt.Connect, d *libvirt.Domain, event *libvirt.DomainEventMigrationIteration) {
+		if vmiHasLocalStorage(vmi) {
+			log.Log.Object(vmi).Warning("Can not migrate VMI with local storage using postcopy, continuing with precopy migration")
+			return
+		}
+
+		if event.Iteration == 1 {
+			log.Log.Object(vmi).Info("Signaled start for post copy migration")
+
+			err := dom.MigrateStartPostCopy(uint32(0))
+			if err != nil {
+				log.Log.Object(vmi).Reason(err).Error("Live postcopy migration failed to start.")
+				migrationError <- err
+				return
+			}
+
+			err = l.updateVMIMigrationMode(dom, vmi, v1.MigrationPostCopy)
+			if err != nil {
+				log.Log.Object(vmi).Reason(err).Error("Unable to update migration mode on domain xml")
+			}
+		}
+	})
+}
+
+func vmiHasLocalStorage(vmi *v1.VirtualMachineInstance) bool {
```

Comment:
I'm a bit confused as to why isn't this allowed?

Reply:
This was done for ease of a first pass implementation

Reply:
I see. Local disks are normally being block-migrated. We calculate whether the vmi should be block-migrated or not during the vmi creation [1] 
As far as I know, block-migration should work with post-copy. However, if these disks are missing from [1], we should just add it :)

[1] https://github.com/kubevirt/kubevirt/blob/a888cef3fac94366b6b5ee71e8496d2c19f9f51e/pkg/virt-handler/vm.go#L1798

Reply:
Is this now being using only in tests? If so, perhaps we should move it to the test itself?

Reply:
I'm wondering, can't we block migrate with the emptyDisk? 
I understand that we have an issue migrating local volumes. However, these local storage volumes will be created with ReadWriteOnce access, so we will prevent the migration of a VMI with such a volume ahead of time in virt-handler. Is my assumption wrong?

Reply:
From my understanding you are correct and this is not needed. I had deleted it, but it looks like I messed up when I rebased. I'll remove the functions, along with the tests for it.